src/Symfony/Component/Translation/CacheTranslatorBag.php

```diff
@@ -0,0 +1,147 @@
+        $this->configCacheFactory = $configCacheFactory ?: new ConfigCacheFactory($this->debug);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCatalogue($locale)
+    {
+        $self = $this; // required for PHP 5.3 where "$this" cannot be use()d in anonymous functions. Change in Symfony 3.0.
+        $cache = $this->configCacheFactory->cache($this->getCatalogueCachePath($locale),
+            function (ConfigCacheInterface $cache) use ($self, $locale) {
+                $self->dumpCatalogue($locale, $cache);
+            }
+        );
+
+        /* Read catalogue from cache. */
+        return include $cache->getPath();
+    }
+
+    /**
+     * @internal
+     */
+    public function initializeCatalogue($locale)
+    {
+        unlink($this->getCatalogueCachePath($locale));
+
+        return $this->getCatalogue($locale);
+    }
+
+    /**
+     * This method is public because it needs to be callable from a closure in PHP 5.3. It should be made protected (or even private, if possible) in 3.0.
+     *
+     * @internal
+     */
+    public function dumpCatalogue($locale, ConfigCacheInterface $cache)
+    {
+        $catalogue = $this->translatorBag->getCatalogue($locale);
+        $fallbackContent = $this->getFallbackContent($catalogue);
+
+        $content = sprintf(<<<EOF
+<?php
+
+use Symfony\Component\Translation\MessageCatalogue;
+
+\$catalogue = new MessageCatalogue('%s', %s);
+
+%s
+return \$catalogue;
+
+EOF
+            ,
+            $locale,
+            var_export($catalogue->all(), true),
+            $fallbackContent
+        );
+
+        $cache->write($content, $catalogue->getResources());
+    }
+
+    private function getFallbackContent(MessageCatalogue $catalogue)
+    {
+        $fallbackContent = '';
+        $current = '';
+        $replacementPattern = '/[^a-z0-9_]/i';
+        $fallbackCatalogue = $catalogue->getFallbackCatalogue();
+        while ($fallbackCatalogue) {
+            $fallback = $fallbackCatalogue->getLocale();
+            $fallbackSuffix = ucfirst(preg_replace($replacementPattern, '_', $fallback));
+            $currentSuffix = ucfirst(preg_replace($replacementPattern, '_', $current));
+
+            $fallbackContent .= sprintf(<<<EOF
+\$catalogue%s = new MessageCatalogue('%s', %s);
+\$catalogue%s->addFallbackCatalogue(\$catalogue%s);
+
+EOF
+                ,
+                $fallbackSuffix,
+                $fallback,
+                var_export($fallbackCatalogue->all(), true),
+                $currentSuffix,
+                $fallbackSuffix
+            );
+            $current = $fallbackCatalogue->getLocale();
+            $fallbackCatalogue = $fallbackCatalogue->getFallbackCatalogue();
+        }
+
+        return $fallbackContent;
+    }
+
+    private function getCatalogueCachePath($locale)
+    {
+        return $this->cacheDir.'/catalogue.'.$locale.'.'.sha1(serialize($this->translatorBag)).'.php';
```

Comment:
Right now I fixed this with quick-fix [Serializable](https://github.com/aitboudad/symfony/blob/translation_loader_resolver/src/Symfony/Component/Translation/TranslatorBag.php#L110) :smiley: 
My solution to that is to create a custom implementation of [ResourceInterface](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Config/Resource/ResourceInterface.php) that preserve the state of TranslatorBag.

```php
class TranslatorBag implements TranslatorBagInterface, \Serializable
{
    /**
     * {@inheritdoc}
     */
    public function getCatalogue($locale)
    {
        ...
        $catalogue->addResource(new TranslatorBagResource($this->resources));
    }
}
```
what do you think ?